plugins/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenPath.java

```diff
@@ -398,7 +398,7 @@ public MavenPath locate(final String extension, @Nullable final String classifie
     String newPath = origin.path.substring(0, origin.path.length() - (origin.coordinates.extension.length() + 1));
     if (origin.coordinates.classifier != null) {
       // strip "-classifier"
-      newPath = newPath.substring(0, newPath.length() - origin.coordinates.classifier.length() + 1);
+      newPath = newPath.substring(0, newPath.length() - (origin.coordinates.classifier.length() + 1));
```

Comment:
Trivial: I think it would make the change clearer if we just swapped the + 1 to a - 1

Reply:
@dohque good point